python/beeai_framework/tools/tool.py

```diff
@@ -231,12 +234,17 @@ def _create_emitter(self) -> Emitter:
                 creator=self,
             )
 
-        async def _run(self, input: Any, options: ToolRunOptions | None, context: RunContext) -> Any:
+        async def _run(self, input: Any, options: ToolRunOptions | None, context: RunContext) -> ToolOutput:
             tool_input_dict = input.model_dump()
             if inspect.iscoroutinefunction(tool_function):
-                return await tool_function(**tool_input_dict)
+                result = await tool_function(**tool_input_dict)
+            else:
+                result = tool_function(**tool_input_dict)
+
+            if isinstance(result, ToolOutput):
+                return result
             else:
-                return tool_function(**tool_input_dict)
+                return StringToolOutput(result=str(result))
```

Comment:
should this also check if result is dict then use JSONToolOutput else default to StringToolOutput?